hw/top_earlgrey/sw/autogen/top_earlgrey.rs

```diff
@@ -0,0 +1,2534 @@
+pub const TOP_EARLGREY_ROM_CTRL_ROM_SIZE_BYTES: usize = 0x8000;
+/// Peripheral base address for cfg device on rv_core_ibex in top earlgrey.
+///
+/// This should be used with #mmio_region_from_addr to access the memory-mapped
+/// registers associated with the peripheral (usually via a DIF).
+pub const TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR: usize = 0x411F0000;
+
+/// Peripheral size for cfg device on rv_core_ibex in top earlgrey.
+///
+/// This is the size (in bytes) of the peripheral's reserved memory area. All
+/// memory-mapped registers associated with this peripheral should have an
+/// address between #TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR and
+/// `TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR + TOP_EARLGREY_RV_CORE_IBEX_CFG_SIZE_BYTES`.
+pub const TOP_EARLGREY_RV_CORE_IBEX_CFG_SIZE_BYTES: usize = 0x100;
+
+/// Memory base address for ram_ret_aon in top earlgrey.
+pub const RAM_RET_AON_BASE_ADDR: usize = 0x40600000;
+
+/// Memory size for ram_ret_aon in top earlgrey.
+pub const RAM_RET_AON_SIZE_BYTES: usize = 0x1000;
+/// Memory base address for eflash in top earlgrey.
+pub const EFLASH_BASE_ADDR: usize = 0x20000000;
+
+/// Memory size for eflash in top earlgrey.
+pub const EFLASH_SIZE_BYTES: usize = 0x100000;
+/// Memory base address for ram_main in top earlgrey.
+pub const RAM_MAIN_BASE_ADDR: usize = 0x10000000;
+
+/// Memory size for ram_main in top earlgrey.
+pub const RAM_MAIN_SIZE_BYTES: usize = 0x20000;
+/// Memory base address for rom in top earlgrey.
+pub const ROM_BASE_ADDR: usize = 0x8000;
+
+/// Memory size for rom in top earlgrey.
+pub const ROM_SIZE_BYTES: usize = 0x8000;
+
+/// PLIC Interrupt Source Peripheral.
+///
+/// Enumeration used to determine which peripheral asserted the corresponding
+/// interrupt.
+pub enum TopEarlgreyPlicPeripheral {
+    /// Unknown Peripheral
+    Unknown = 0,
+    /// uart0
+    Uart0 = 1,
+    /// uart1
+    Uart1 = 2,
+    /// uart2
+    Uart2 = 3,
+    /// uart3
+    Uart3 = 4,
+    /// gpio
+    Gpio = 5,
+    /// spi_device
+    SpiDevice = 6,
+    /// i2c0
+    I2c0 = 7,
+    /// i2c1
+    I2c1 = 8,
+    /// i2c2
+    I2c2 = 9,
+    /// pattgen
+    Pattgen = 10,
+    /// rv_timer
+    RvTimer = 11,
+    /// otp_ctrl
+    OtpCtrl = 12,
+    /// alert_handler
+    AlertHandler = 13,
+    /// spi_host0
+    SpiHost0 = 14,
+    /// spi_host1
+    SpiHost1 = 15,
+    /// usbdev
+    Usbdev = 16,
+    /// pwrmgr_aon
+    PwrmgrAon = 17,
+    /// sysrst_ctrl_aon
+    SysrstCtrlAon = 18,
+    /// adc_ctrl_aon
+    AdcCtrlAon = 19,
+    /// aon_timer_aon
+    AonTimerAon = 20,
+    /// sensor_ctrl
+    SensorCtrl = 21,
+    /// flash_ctrl
+    FlashCtrl = 22,
+    /// hmac
+    Hmac = 23,
+    /// kmac
+    Kmac = 24,
+    /// otbn
+    Otbn = 25,
+    /// keymgr
+    Keymgr = 26,
+    /// csrng
+    Csrng = 27,
+    /// entropy_src
+    EntropySrc = 28,
+    /// edn0
+    Edn0 = 29,
+    /// edn1
+    Edn1 = 30,
+    /// \internal Final PLIC peripheral
+    Last = 30,
```

Comment:
I don't think rust is going to allow a name that is assigned a duplicated discriminant (the value 30 here).

Reply:
I think it will be safe to rename it to End and increment value (End = last entry + 1).